### PR TITLE
feat: Wpp Business's Labels support

### PR DESCRIPTION
**Pull request by dyegoaurelio**
We're now able to work with Wpp Business's Labels

I added methods to get all labels, get all labels a chat is marked and to fetch all chats a Label is marking

### CODE REVIEW

File: src/Client.js [hunk 1]
```diff
@@ -741,6 +741,41 @@ class Client extends EventEmitter {
         return { gid: createRes.gid, missingParticipants };
     }
 
+    /**
+     * Returns all current Labels
+     * @returns {Promise<Array<Label>>}
+     */
+    async getLabels() {
+        return JSON.parse( await this.pupPage.evaluate(async () => {
```
Comment (pedroslopez) [inline src/Client.js:749]: No need to encode/decode JSON here. Puppeteer takes care of serializing the models.
Also, no need to `return await`. Opt for `return` instead. The https://eslint.org/docs/rules/no-return-await rule will be enabled on eslint soon.

File: src/Client.js [hunk 2]
```diff
@@ -741,6 +741,41 @@ class Client extends EventEmitter {
...
+            return JSON.stringify(await window.WWebJS.getLabels());
+        }) );
+    }
+
+    /**
+     * Returns Label of specified chat 
+     * @returns {Promise<Label>}
+     */
+    async getChatLabels(chatId){
+        return JSON.parse( await this.pupPage.evaluate(async (chatId) => {
```
Comment (pedroslopez) [inline src/Client.js:759]: Same as before, JSON conversion and no `return await`.

File: src/Client.js [hunk 3]
```diff
@@ -741,6 +741,41 @@ class Client extends EventEmitter {
...
+            return JSON.stringify(await window.WWebJS.getChatLabels(chatId));
+        }, chatId) );
+    }
+
+    /**
+     * Returns all Chats of specified label
+     * @returns {Array<ChatId>}
+     */
+    async getAllChatsFromLabel(labelId){
+        return ( await this.pupPage.evaluate(async (labelId) => {
```
Comment (pedroslopez) [inline src/Client.js:769]: Unneeded `return await`

File: src/structures/Chat.js [hunk 4]
```diff
@@ -228,6 +228,14 @@ class Chat extends Base {
     async getContact() {
         return await this.client.getContactById(this.id._serialized);
     }
+
+    /**
+     * Returns array of all labels this Chat is marked
+     * @returns {Array<Labels>}
+     */
+    async getLabels() {
+        return await this.client.getChatLabels(this.id._serialized);
```
Comment (pedroslopez) [inline src/structures/Chat.js:237]: Unneeded `return await`

File: src/util/Injected.js [hunk 5]
```diff
@@ -317,6 +317,21 @@ exports.LoadUtils = () => {
         return true;
     };
 
+    window.WWebJS.getLabels = async () =>{
+        return window.Store.Label.models.map( function (label){  return  { 'name': label.name , 'hexColor': label.hexColor , 'id':label.id }; });
```
Comment (pedroslopez) [inline src/util/Injected.js:321]: Prefer arrow functions here to keep consistent with the rest of the code.

File: src/util/Injected.js [hunk 6]
```diff
@@ -317,6 +317,21 @@ exports.LoadUtils = () => {
...
+    };
+
+    window.WWebJS.getChatLabels = async (chatId) => {
+        let chatLabelsIndex = await window.WWebJS.getChat(chatId);
+        chatLabelsIndex = chatLabelsIndex.labels;
+        //console.log(chatLabelsIndex)
```
Comment (pedroslopez) [inline src/util/Injected.js:327]: Remove commented code

File: src/util/Injected.js [hunk 7]
```diff
@@ -317,6 +317,21 @@ exports.LoadUtils = () => {
...
+        let res = [];
+        for (var i in chatLabelsIndex){
+            res.push({ 'name': window.Store.Label.get(chatLabelsIndex[i]).name , 'hexColor': window.Store.Label.get(chatLabelsIndex[i]).hexColor ,  'id' : chatLabelsIndex[i] } );
```
Comment (pedroslopez) [inline src/util/Injected.js:330]: I'm seeing in multiple places you're constructing this object representation of a Label with the same properties. To keep things consistent and remove code duplication, perhaps we could instead have a serializer function, much like we have with `getChatModel` and `getMessageModel` that takes care of returning a properly serialized representation of the label with the necessary properties.

**Review by pedroslopez — CHANGES_REQUESTED** [review]
Thanks for contributing! Sorry it took a while to review.

Looks good overall and works functionality-wise. However, I believe instead of returning regular objects for labels in functions like `client.getChatLabels()` and `client.getLabels()`, we should opt for creating a Class that represents a ChatLabel. This way, we have a structured representation of labels and their properties that can be properly documented. It could also be useful if we want to add functions like editing labels in the future.

**Comment by dyegoaurelio** [comment]
Closed this pr accidentally
reopened at #407

**Pull request closed without merging**